### PR TITLE
philadelphia-core: Improve 'FIXValue#asFloat' and 'FIXValue#setFloat'

### PR DESCRIPTION
**Pull request by jvirtanen**

### CODE REVIEW

File: libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java [hunk 1]
```diff
@@ -211,48 +232,87 @@ public void setInt(long x) {
...
 
-            x = 10 * x + b - '0';
+            significand = 10 * significand + b - '0';
 
-            factor *= 10;
+            exponent++;
         }
 
-        return sign * (factor > 0.0 ? x / factor : x);
+        double x = exponent != 0 ? significand / POWERS_OF_TEN[exponent] : significand;
```
Comment (jheusser) [inline libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java]: Could you drop this conditional and just write below as `POWERS_OF_TEN[0] == 1`:
```
double x = significand / POWERS_OF_TEN[exponent];
```

Maybe something for the benchmark to decide.

Comment (jvirtanen) [reply libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java]: Good catch! 🔍  We certainly don't want to optimize for `exponent == 0` at the expense of `exponent != 0`.

Comment (jvirtanen) [reply libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java]: This is now fixed. 👍

File: libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java [hunk 2]
```diff
@@ -211,48 +232,87 @@ public void setInt(long x) {
...
+            significand = 10 * significand + b - '0';
 
-            factor *= 10;
+            exponent++;
         }
 
-        return sign * (factor > 0.0 ? x / factor : x);
+        double x = significand / POWERS_OF_TEN[exponent];
+
+        return negative ? -x : +x;
```
Comment (jheusser) [inline libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java]: What about keeping this branchless?

Comment (jvirtanen) [reply libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java]: I don't see a perceptible difference between this and a branchless version, but then again the performance test only tests positive numbers. 🙂 On the other hand, I can imagine many applications will generally also predominantly if not exclusively work with positive numbers.

In the future, I will look into some related enhancements in the performance tests: positive and negative numbers, numbers with different number of digits and so on.

**Pull request merged**